src/cloudevents/core/v1/event.py

```diff
@@ -0,0 +1,140 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from typing import Any, Optional
+from datetime import datetime
+import re
+
+REQUIRED_ATTRIBUTES = {"id", "source", "type", "specversion"}
+OPTIONAL_ATTRIBUTES = {"datacontenttype", "dataschema", "subject", "time"}
+
+
+class CloudEvent:
+    def __init__(self, attributes: dict, data: Optional[dict] = None) -> None:
+        """
+        Create a new CloudEvent instance.
+
+        :param attributes: The attributes of the CloudEvent instance.
+        :type attributes: dict
+        :param data: The payload of the CloudEvent instance.
+        :type data: Optional[dict]
+
+        :raises ValueError: If any of the required attributes are missing or have invalid values.
+        :raises TypeError: If any of the attributes have invalid types.
+        """
+        self._validate_attribute(attributes)
+        self._attributes = attributes
+        self._data = data
+
+    def _validate_attribute(self, attributes: dict) -> None:
```

Comment:
there's no need to have this method as an instance method. Let's convert to `static`. It'd also allow to test it separately and easier if needed